### PR TITLE
Implementing JWT authentication

### PR DESCRIPTION
**Pull request by mariuskriukovas**

### CODE REVIEW

**Review by Trutovik — COMMENTED** [review]
Please add the exceptions handler, I would suggest @ControllerAdvice annotation

File: accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/service/AuthorizationService.java [hunk 1]
```diff
@@ -1,7 +1,8 @@
 package io.marius.demo.ecommerce.accountservice.service;
 
 import io.marius.demo.ecommerce.accountservice.model.payload.LoginPayload;
+import org.springframework.http.ResponseEntity;
```
Comment (Trutovik) [inline accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/service/AuthorizationService.java:5]: you don't need this interface unless you want to perform inheritance with more than 1 implementation

File: accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/controller/SessionController.java [hunk 2]
```diff
@@ -5,17 +5,18 @@
...
-@RequestMapping("sessions/")
+@RequestMapping("sessions")
 @RequiredArgsConstructor
 @FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
 public class SessionController {
   AuthorizationService authorizationService;
 
   @PostMapping("login")
-  String login(@RequestBody LoginPayload payload) {
+  ResponseEntity<String> login(@RequestBody LoginPayload payload) {
```
Comment (Trutovik) [inline accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/controller/SessionController.java:19]: add validation here, at least @NotNull annotations in your entity

File: accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/controller/SessionController.java [hunk 3]
```diff
@@ -5,17 +5,18 @@
...
 
 @RestController
-@RequestMapping("sessions/")
+@RequestMapping("sessions")
 @RequiredArgsConstructor
 @FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
 public class SessionController {
   AuthorizationService authorizationService;
 
   @PostMapping("login")
```
[the remark this answers is not in the document]
Comment (mariuskriukovas) [reply accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/controller/SessionController.java:18]: SessionController has been extended with BaseController form the commons-api module and now inherits a getHealth method.

File: accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/controller/SessionController.java (hunk 2 above)
Comment (mariuskriukovas) [reply accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/controller/SessionController.java]: Added **@Validated** annotation to enforce validation rules for @NotBlank and @NotNull annotations.

Comment (mariuskriukovas) [reply accountservice/src/main/java/io/marius/demo/ecommerce/accountservice/controller/SessionController.java]: A new commons-validation package has been created, which includes a GlobalExceptionHandler that can handle MethodArgumentNotValidException globally in the application

**Pull request merged**